### PR TITLE
[ir] Remove Matrix Kind from the IR

### LINKED ISSUE #229: Remove `kind` from the matrix type variants
Apparently, SPIR-V only works with floating-point matrices!

### PR DESCRIPTION
**Pull request by Napokue**
Should fix #229. I haven't included any additional validation, as I don't know how to do that (yet). I don't know if that is necessary for this PR or if that can be added later.

### CODE REVIEW

File: src/back/glsl.rs [hunk 1]
```diff
@@ -1405,20 +1404,15 @@ fn write_type<'a>(
         TypeInner::Matrix {
             columns,
             rows,
-            kind,
             width,
         } => {
-            if kind != ScalarKind::Float {
```
Comment (kvark) [inline src/back/glsl.rs:1411]: oh, nice!

File: src/proc/validator.rs [hunk 2]
```diff
@@ -228,14 +228,18 @@ impl Validator {
...
-                            | crate::TypeInner::Matrix { kind, .. } => {
+                            | crate::TypeInner::Vector { kind, .. } => {
                                 if kind != crate::ScalarKind::Float
                                     && var.interpolation != Some(crate::Interpolation::Flat)
                                 {
                                     return Err(EntryPointError::InvalidIntegerInterpolation);
                                 }
                             }
+                            crate::TypeInner::Matrix { .. } => {
+                                if var.interpolation != Some(crate::Interpolation::Flat) {
```
Comment (kvark) [inline src/proc/validator.rs:239]: this condition should be removed from here, the code above checked "kind != crate::ScalarKind::Float &&  this", so in this branch it's guaranteed false

**Review by kvark — CHANGES_REQUESTED** [review]
Thank you!

File: src/proc/validator.rs (hunk 2 above)
Comment (Napokue) [reply src/proc/validator.rs:239]: I'm sorry, but I have no idea what interpolation means. So I can remove this entire match arm? As this is the only check.

Comment (kvark) [reply src/proc/validator.rs:239]: This would essentially be `crate::TypeInner::Matrix { .. } => {}`

**Review by kvark — APPROVED** [review]

**Pull request merged**